### DISCUSSION TITLE
Do Jedis support RFC 6555 for dual-stack? A.K.A happy eyeballs.

### DISCUSSION THREAD

**Discussion by jonatiao** [xetorthio/jedis#3441] [Q&A]
Hello Everyone, first of all thx for your work on Jedis!

We are using redis.clients:jedis:jar:3.8.0.

Some servers has both IPV4(A) and IPV6(AA) addresses, this is called dual-stack and it's normal for most of the cloud resources.
Sometimes both addresses responde fine.. but sometimes, most of the time the ipv6 address does not. So they "invented" the happy eyeballs, which in summary, fallback fast to ipv4 when a problem occurs.

The recommendation for clients that consume those dual stack services is to implement RFC6555.

I'm reaching you guys because I see this behavior, sometimes, not too rare, in our services when connecting to azure cache for redis we get a long wait for ipv6 and we don't see the fallback to ipv4.

For instance, as an example, Firefox has this implemented as a "about:config" network.http.fast-fallback-to-IPv4.

Jedis has anything like it?

thx!

**Comment by sazzad16** [comment]
@jonatiao Not exactly. BUT but default Jedis uses `java.net.InetAddress`, `java.net.InetSocketAddress` and `java.net.Socket`. If Java/JDK has any configuration to alter their behavior system wide, that will also effect Jedis sockets. (I haven't found any such thing.)

**Reply by jonatiao** [reply]
Hey @sazzad16, thx for your answer!! really appreciated. ^^

I believe java/jdk does not implement it. This explains why we have netty and okhttp trying to:

Netty is trying: https://github.com/netty/netty/issues/9540
okhttp is trying: https://square.github.io/okhttp/changelogs/changelog/

Can I ask a feature request for Jedis? It's fine if you say no. I do not want to ask anything that does not make sense for your project.

thx again for your answer.

**Reply by sazzad16** [reply]
@jonatiao You can ask for a feature request. I can't promise it'll be worked on. But if we get a PR, we'll definitely look into that.
